azurerm/internal/services/containers/kubernetes_addons.go

```diff
@@ -1,24 +1,42 @@
 package containers
 
 import (
+	"fmt"
 	"strings"
 
 	"github.com/Azure/azure-sdk-for-go/services/containerservice/mgmt/2020-02-01/containerservice"
+	"github.com/Azure/go-autorest/autorest/azure"
 	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
 	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
-	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	azureHelpers "github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
 	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
 )
 
 const (
+	// note: the casing on these keys is important
 	aciConnectorKey           = "aciConnectorLinux"
 	azurePolicyKey            = "azurepolicy"
 	kubernetesDashboardKey    = "kubeDashboard"
 	httpApplicationRoutingKey = "httpApplicationRouting"
 	omsAgentKey               = "omsagent"
 )
 
-func SchemaKubernetesAddOnProfiles() *schema.Schema {
+// The AKS API hard-codes which add-ons are supported in which environment
+// as such unfortunately we can't just send "disabled" - we need to strip
+// the unsupported addons from the HTTP response. As such this defines
+// the list of unsupported addons in the defined region - e.g. by being
+// omited from this list an addon/environment combination will be supported
+var unsupportedAddonsForEnvironment = map[string][]string{
+	azure.ChinaCloud.Name: {
+		aciConnectorKey,           // https://github.com/terraform-providers/terraform-provider-azurerm/issues/5510
+		httpApplicationRoutingKey, // https://github.com/terraform-providers/terraform-provider-azurerm/issues/5960
+	},
+	azure.USGovernmentCloud.Name: {
+		httpApplicationRoutingKey, // https://github.com/terraform-providers/terraform-provider-azurerm/issues/5960
+	},
```

Comment:
I went digging through the Docs & AKS Engine to try and find a definitive list of which add-ons - but ended up with this based on then open issues

@jluk do you know if there's a document stating which add-ons are supported in which cloud environments?

Reply:
Unfortunately not today.

@feiskeyer to confirm the 2 missing addons in Azure China/Gov.

Reply:
there're some list under https://github.com/Azure/AKS/projects/1